JamGame/Assets/Scripts/TileBuilder/TileBuilderImpl.cs

```diff
@@ -388,5 +403,38 @@ public BuildingConfig SaveBuildingIntoConfig()
 
             return buildingConfig;
         }
+
+        public List<CoreModel> ClearForGrowingMeetingRoom(
+            MeetingRoomGrowingInformation meetingRoomGrowingInformation
+        )
+        {
+            List<CoreModel> coreModels = new();
+
+            IEnumerable<Vector2Int> positionToBorrow =
+                meetingRoomGrowingInformation.PositionsToTake.Where(
+                    x => GetTileUnionInPosition(x).GetAllUniqueMarks().All(x => x != "Freespace")
+                );
+
+            foreach (Vector2Int position in positionToBorrow)
+            {
+                BorrowRoom command = new(position);
+                _ = ExecuteCommand(command);
+                coreModels.Add(command.BorrowedRoom);
+            }
+
+            foreach (Vector2Int position in meetingRoomGrowingInformation.PositionsToTake)
+            {
+                _ = DeleteTile(GetTileUnionInPosition(position));
+            }
+
+            RemoveTileFromDictionary(meetingRoomGrowingInformation.TargetTileUnion);
+            return coreModels;
+        }
+
+        public void PlaceBackMeetingRoom(TileUnionImpl tileUnionImpl)
```

Comment:
```suggestion
        public void AddTileUnion(TileUnionImpl tileUnionImpl)
```
If you don't already have such a method